openforcefield/tests/test_molecule.py

```diff
@@ -1811,28 +1814,31 @@ def test_add_bond_charge_virtual_site(self, molecule):
 
         # Try to feed in a unitless distance
         with pytest.raises(AssertionError) as excinfo:
-            vsite1_index = molecule.add_bond_charge_virtual_site([atom1, atom2, atom3], distance_unitless)
+            vsite1_index = molecule.add_bond_charge_virtual_site([atom1, atom2], distance_unitless)
 
 
-        vsite1_index = molecule.add_bond_charge_virtual_site([atom1, atom2, atom3], distance)
+        vsite1_index = molecule.add_bond_charge_virtual_site([atom1, atom2], distance)
         vsite1 = molecule.virtual_sites[vsite1_index]
         assert atom1 in vsite1.atoms
         assert atom2 in vsite1.atoms
-        assert atom3 in vsite1.atoms
         assert vsite1 in atom1.virtual_sites
         assert vsite1 in atom2.virtual_sites
-        assert vsite1 in atom3.virtual_sites
         assert vsite1.distance == distance
 
         # Make an "everything bagel" virtual site
-        vsite2_index = molecule.add_bond_charge_virtual_site([atom1, atom2, atom3],
+        vsite2_index = molecule.add_bond_charge_virtual_site([atom2, atom3],
                                                              distance,
                                                              sigma=0.1*unit.angstrom,
                                                              epsilon=1.0*unit.kilojoule_per_mole,
-                                                             charge_increments=unit.Quantity(np.array([0.1, 0.2, 0.3]),
+                                                             charge_increments=unit.Quantity(np.array([0.1, 0.2]),
                                                                                              unit.elementary_charge)
                                                              )
         vsite2 = molecule.virtual_sites[vsite2_index]
+        assert atom2 in vsite2.atoms
+        assert atom3 in vsite2.atoms
+        assert vsite2 in atom2.virtual_sites
+        assert vsite2 in atom3.virtual_sites
+        assert vsite2.distance == distance
```

Comment:
Agree that this would be a good thing to add -- like a test to ensure that vsite setters do something sane with Quantities that have the wrong units (maybe raise an error), and to make sure that the getter returns a value that was just set.

These are kinda trivial tests and are really unlikely to actually catch a problem, so it's OK with me to leave them as TODO items if you get short on time.

Reply:
I actually think the Quantity tests were already implemented previously, so I just had to tidy them up a bit.